src/game/Strategic/Strategic_AI.cc

```diff
@@ -3139,7 +3140,7 @@ static void RequestHighPriorityGarrisonReinforcements(size_t iGarrisonID, UINT8
 	}
 	ubDstSectorX = (UINT8)SECTORX( gGarrisonGroup[ iGarrisonID ].ubSectorID );
 	ubDstSectorY = (UINT8)SECTORY( gGarrisonGroup[ iGarrisonID ].ubSectorID );
-	if( iBestIndex != -1 )
+	if( iBestIndex != (size_t)-1 )
```

Comment:
How does it work? Won't the cast change `-1` into `INT_MAX`? Can it still compare to a signed int?

Reply:
Yes, as I understand it it should be something like SIZE_MAX which is OK if "size_t iBestIndex = -1;" is equivalent to "size_t iBestIndex = SIZE_MAX;" There are other cases where -1 was inteded to returned as SIZE_MAX (but specified only as -1) to flag a special case such as in "size_t const patrol_id = FindPatrolGroupIndexForGroupID(g.ubGroupID);"

Maybe, as you suggest, we should be using the limits, such as SIZE_MAX in this case, to be clear on the intention.

Reply:
Aha. I didn't realize `iBestIndex` is unsigned